### PR TITLE
WIRE-839: add service account keys support

### PR DESCRIPTION
**Pull request by radekska**
- resource implementation for `castai_service_account_key` 
- minor tweaks for `castai_service_account` resource

### CODE REVIEW

File: castai/resource_service_account.go [hunk 1]
```diff
@@ -278,3 +295,276 @@ func getOrganizationID(ctx context.Context, data *schema.ResourceData, meta inte
 
 	return organizationID, nil
 }
+
+func resourceServiceAccountKey() *schema.Resource {
```
Comment (oskarwojciski) [inline castai/resource_service_account.go:299]: If we treat this as separate resource, why not make a separate file `resource_service_account_key.go` to keep the files smaller?

**Review by oskarwojciski — COMMENTED** [review]
Can you add some example of working with keys, also how to obtain the key token value while creating it?

I wonder if we want to split the resources or make the keys part of the `resource_service_account`. They are separated in the API, but logically, they are very much connected.

File: castai/resource_service_account.go [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+				Required:         true,
+				Description:      "Name of the service account.",
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+			},
+			FieldServiceAccountDescription: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the service account.",
+			},
+			FieldServiceAccountEmail: {
```
Comment (Vizualni) [inline castai/resource_service_account.go:74]: Do we also need to provide the service account id field?

**Comment by radekska** [comment]
> Can you add some example of working with keys, also how to obtain the key token value while creating it?
>
Examples added

> I wonder if we want to split the resources or make the keys part of the `resource_service_account`. They are separated in the API, but logically, they are very much connected.

I thought about the same and decided to follow GCP approach:
- https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_service_account
- https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_service_account_key

**Comment by oskarwojciski** [comment]
I see that we show change state for the name change, but we don't allow to update it, so it never apply
image
should we add `ForceNew` flag for `name`, and the `service_account_id` fields? Same as for `expires_at` which works fine

File: castai/resource_service_account.go (hunk 2 above)
Comment (radekska) [reply castai/resource_service_account.go]: Could you elaborate on that? I'm not sure what you meant

**Pull request merged**